### PR TITLE
Buildable shuttle propulsion and access changes

### PR DESCRIPTION
**Pull request by XElectricX**
Built using an igniter, cable coil, 5 capacitors, 20 osmium sheets, and 10 uranium sheets.

### CODE REVIEW

**Comment by XElectricX** [comment]
Mkay, fuck, I guess just making it buildable like a machine will not work. I'll think of a solution.

File: code/game/shuttle_engines.dm [hunk 1]
```diff
@@ -32,6 +32,57 @@
 	icon_state = "propulsion"
 	opacity = 1
 
+//Working like regular machines until a better way to make them is made
+/obj/structure/shuttle/engine/propulsion/New()
+	..()
+	component_parts = list()
```
Comment (Zero-Bits) [inline code/game/shuttle_engines.dm]: Only machinery has component parts.

Comment (Zero-Bits) [reply code/game/shuttle_engines.dm]: `code/game/shuttle_engines.dm:78:error: component_parts: undefined var`

File: code/game/shuttle_engines.dm [hunk 2]
```diff
@@ -32,6 +32,57 @@
...
+	component_parts += new /obj/item/stack/material/ocp(src)
+	component_parts += new /obj/item/stack/material/ocp(src)
+	component_parts += new /obj/item/stack/material/ocp(src)
+	component_parts += new /obj/item/stack/material/ocp(src)
+	component_parts += new /obj/item/stack/material/ocp(src)
+	component_parts += new /obj/item/stack/material/ocp(src)
+	component_parts += new /obj/item/stack/material/ocp(src)
+	component_parts += new /obj/item/stack/material/ocp(src)
+	component_parts += new /obj/item/stack/material/ocp(src)
+	RefreshParts()
```
Comment (Zero-Bits) [inline code/game/shuttle_engines.dm]: `code/game/shuttle_engines.dm:117:error: RefreshParts: undefined proc`

These vars and procs don't exist for structure objects.

**Pull request merged**